lib/LLVMIRCodeGen/AllocationsInfo.cpp

```diff
@@ -51,8 +90,29 @@ void AllocationsInfo::allocateWeightVars(const IRFunction *F) {
     allocatedAddress_[w] = addr;
   }
 
-  // Compute the offsets and total memory requirements for Placeholders.
+  // Placeholders order should match the allocation order in BackendUtils.
+  std::vector<Placeholder const *> inputOutputPlaceholders;
+  std::vector<Placeholder const *> outputPlaceholders;
+  std::vector<Placeholder const *> contiguousPlaceholders;
   for (auto &v : F->findPlaceholders()) {
+    if (isInputPlaceholder(v)) {
```

Comment:
Isn't this a copy of the code above? Can we commonalize with a helper function instead?

Reply:
> Isn't this a copy of the code above? Can we commonalize with a helper function instead?

OK.